### PR TITLE
Add possibility to unassign cce_node eip

### PR DESCRIPTION
**Pull request by lego963**
## Summary of the Pull Request
This pr contains workaround of eip issue with cce_node_v3. It can only diassociate eip.

## PR Checklist

* [x] Refers to: #678 
* [x] Tests added/passed.
* [ ] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
=== RUN   TestAccCCENodesV3_basic
--- PASS: TestAccCCENodesV3_basic (1929.79s)
=== RUN   TestAccCCENodesV3_timeout
--- PASS: TestAccCCENodesV3_timeout (1853.92s)
=== RUN   TestAccCCENodesV3_ip
--- PASS: TestAccCCENodesV3_ip (1886.24s)
=== RUN   TestAccCCENodesV3_ipWithExtendedParameters
--- PASS: TestAccCCENodesV3_ipWithExtendedParameters (1868.94s)
=== RUN   TestAccCCENodesV3_ipNulls
--- PASS: TestAccCCENodesV3_ipNulls (1871.13s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go [hunk 1]
```diff
@@ -654,6 +640,19 @@ func resourceCCENodeV3Update(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	// release ip if assigned
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go]: And if other EIP params has changed?

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go [hunk 2]
```diff
@@ -654,6 +640,19 @@ func resourceCCENodeV3Update(d *schema.ResourceData, meta interface{}) error {
...
+	if d.HasChange("bandwidth_size") {
+		_, newBandWidthSize := d.GetChange("bandwidth_size")
+		newBandWidth := newBandWidthSize.(int)
+
+		if newBandWidth == 0 {
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go:618]: And if bandwidth changed, e.g. from 10 to 100?

**Review by outcatcher — CHANGES_REQUESTED** [review]

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go [hunk 3]
```diff
@@ -654,6 +642,31 @@ func resourceCCENodeV3Update(d *schema.ResourceData, meta interface{}) error {
...
+		}
+		if newBandWidth == 0 && oldBandwidth > 0 {
+			if err := resourceCCENodeV3DeleteAssociateIP(d, config, serverId, floatingIp); err != nil {
+				return err
+			}
+		} else if newBandWidth > 0 && oldBandwidth > 0 {
+			if err := resourceCCENodeV3ResizeBandwidth(d, config, floatingIp.ID, newBandWidth); err != nil {
+				return err
+			}
+		} else if newBandWidth > 0 && floatingIp == nil {
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go]: What is the goal for ` && floatingIp == nil` check?

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go [hunk 4]
```diff
@@ -654,6 +642,31 @@ func resourceCCENodeV3Update(d *schema.ResourceData, meta interface{}) error {
...
+		oldBandwidthSize, newBandWidthSize := d.GetChange("bandwidth_size")
+		newBandWidth := newBandWidthSize.(int)
+		oldBandwidth := oldBandwidthSize.(int)
+		serverId := d.Get("server_id").(string)
+		floatingIp, err := getCCENodeV3FloatingIp(d, meta, serverId)
+		if err != nil {
+			return err
+		}
+		if newBandWidth == 0 && oldBandwidth > 0 {
+			if err := resourceCCENodeV3DeleteAssociateIP(d, config, serverId, floatingIp); err != nil {
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go]: You can check the error after the end of the `if... else if ..` block

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3_test.go [hunk 5]
```diff
@@ -71,6 +71,39 @@ func TestAccCCENodesV3_ip(t *testing.T) {
 					resource.TestCheckResourceAttr("opentelekomcloud_cce_node_v3.node_1", "bandwidth_charge_mode", "traffic"),
 				),
 			},
+			{
+				Config: testAccCCENodeV3_bandWidthResize,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCCENodeV3Exists("opentelekomcloud_cce_node_v3.node_1", "opentelekomcloud_cce_cluster_v3.cluster_1", &node),
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3_test.go:78]: Please add check for bandwidth size

**Review by outcatcher — CHANGES_REQUESTED** [review]

**Review by anton-sidelnikov — DISMISSED** [review]

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go [hunk 6]
```diff
@@ -686,6 +715,110 @@ func resourceCCENodeV3Delete(d *schema.ResourceData, meta interface{}) error {
...
+	createEipOpts := EIPCreateOpts{
+		ApplyOpts: eips.ApplyOpts{
+			IP: eips.PublicIpOpts{
+				Type: "5_bgp",
+			},
+			Bandwidth: eips.BandwidthOpts{
+				Name:       "cce-node-1",
+				Size:       size,
+				ShareType:  "PER",
+				ChargeMode: "traffic",
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go]: Only `ShareType` is a required field. Why do you hardcode optional parameters?

Comment (outcatcher) [reply opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go]: Also, where possible, you should take values from the schema

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by outcatcher — DISMISSED** [review]

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3_test.go [hunk 7]
```diff
@@ -83,6 +83,30 @@ func TestAccCCENodesV3_bandWidthResize(t *testing.T) {
...
+				),
+			},
+			{
+				Config: testAccCCENodeV3_ipUnset,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCCENodeV3Exists("opentelekomcloud_cce_node_v3.node_1", "opentelekomcloud_cce_cluster_v3.cluster_1", &node),
+				),
+			},
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3_test.go]: Can you please add a step with assigning EIPs back

File: opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go [hunk 8]
```diff
@@ -654,6 +635,49 @@ func resourceCCENodeV3Update(d *schema.ResourceData, meta interface{}) error {
...
+		newEipIds := newEipIdsRaw.(*schema.Set).List()
+		serverId := d.Get("server_id").(string)
+		if len(newEipIds) == 0 {
+			if err := deleteCCENodeV3AssociatedIP(d, config, serverId, oldEipIds[0].(string)); err != nil {
+				return err
+			}
+		} else if len(oldEipIds) > 0 {
+			err = reassignCCENodeV3Eip(d, meta, oldEipIds[0].(string), newEipIds[0].(string), serverId)
+		}
+
```
Comment (outcatcher) [inline opentelekomcloud/resource_opentelekomcloud_cce_node_v3.go]: You can find removed/added `eip_ids` with `schema.Set.Difference`

**Review by anton-sidelnikov — DISMISSED** [review]

**Comment by outcatcher** [comment]
Postponed to v1.22 due to CCE  testing problems

**Review by outcatcher — APPROVED** [review]

**Pull request merged**